js/mexc3.js

```diff
@@ -0,0 +1,3659 @@
+            //             "orderListId": "-1",
+            //             "clientOrderId": "",
+            //             "price": "45000",
+            //             "origQty": "0.0002",
+            //             "executedQty": "0",
+            //             "cummulativeQuoteQty": "0",
+            //             "status": "NEW",
+            //             "timeInForce": null,
+            //             "type": "LIMIT",
+            //             "side": "SELL",
+            //             "stopPrice": null,
+            //             "icebergQty": null,
+            //             "time": "1647718255199",
+            //             "updateTime": null,
+            //             "isWorking": true,
+            //             "origQuoteOrderQty": "9"
+            //         }
+            //     ]
+            //
+            return this.parseOrders (response, market, since, limit);
+        } else {
+            // TO_DO: another possible way is through: open_orders/{symbol}, but as they have same ratelimits, and less granularity, i think historical orders are more convenient, as it supports more params (however, theoretically, open-orders endpoint might be sligthly fast)
+            return await this.fetchOrdersByState (2, symbol, since, limit, params);
+        }
+    }
+
+    async fetchClosedOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        return await this.fetchOrdersByState (3, symbol, since, limit, params);
+    }
+
+    async fetchCanceledOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        return await this.fetchOrdersByState (4, symbol, since, limit, params);
+    }
+
+    async fetchOrdersByState (state, symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        const [ marketType, query ] = this.handleMarketTypeAndParams ('fetchOrdersByState', undefined, params);
+        await this.loadMarkets ();
+        const request = {};
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+            request['symbol'] = market['id'];
+        }
+        if (marketType === 'spot' || (market && market['spot'])) {
+            throw new BadRequest (this.id + ' fetchOrdersByState() is not supported for ' + marketType);
+        } else {
+            query['states'] = state;
+            return this.fetchOrders (symbol, since, limit, query);
+        }
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        const [ marketType, query ] = this.handleMarketTypeAndParams ('cancelOrder', undefined, params);
+        await this.loadMarkets ();
+        const request = {};
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+            request['symbol'] = market['id'];
+        }
+        let data = undefined;
+        if (marketType === 'spot' || (market && market['spot'])) {
```

Comment:
As suggested before, If you replace all occurrences of 

```
const [ marketType, query ] = this.handleMarketTypeAndParams ('methodName', undefined, params);
````
by

```Javascript
const market = symbol !== undefined this.symbol(market) : undefined
const [ marketType ] = this.handleMarketTypeAndParams ('methodName', market, params);
```
then you can simply replace all of these conditions  by:

```Javascript
if (marketType !== 'spot) {
(...)
}
```

Reply:
good note! you can go on with them